src/components/App.tsx

```diff
@@ -269,19 +270,31 @@ export class App extends React.Component<any, AppState> {
     if (this.unmounted) {
       return;
     }
+
+    let editingElement: AppState["editingElement"] | null = null;
     if (res.elements) {
+      res.elements.forEach((element) => {
+        if (
+          this.state.editingElement?.id === element.id &&
```

Comment:
@dwelle can you explain a bit about this and how it works? We could add some comments as well.

Reply:
the idea here is that if the action recreated the element, the referential integrity of `state.editingElement` would be lost (it'd point to the old element). We want to keep it up to date.

We should do this for other state variables that point to elements, such as `state.resizingElement` etc., but I didn't do it since it's unrelated to the PR and also I'm not sure if this is the way we wanna do it.

Reply:
@dwelle does the `editingElement` represents only a text element or other elements as well? If only text element, could we possibly merge the `editingElement` with the `wysiwigElement` and only keep one?

Reply:
No, it's also `ExcalidrawLinearElement` (line/arrow), and later it'll possibly include all of them.